webknossos-datastore/app/com/scalableminds/webknossos/datastore/storage/httpsfilesystem/HttpsSeekableByteChannel.scala

```diff
@@ -21,46 +20,85 @@ class HttpsSeekableByteChannel(path: HttpsPath, openOptions: util.Set[_ <: OpenO
   private val connectionTimeout = 5 seconds
   private val readTimeout = 1 minute
 
-  private val responseBox: Box[HttpResponse[Array[Byte]]] = tryo(
-    path.getBasicAuthCredential.map { credential =>
-      Http(uri.toString)
-        .timeout(connTimeoutMs = connectionTimeout.toMillis.toInt, readTimeoutMs = readTimeout.toMillis.toInt)
-        .auth(credential.username, credential.password)
-        .asBytes
-    }.getOrElse(Http(uri.toString)
-      .timeout(connTimeoutMs = connectionTimeout.toMillis.toInt, readTimeoutMs = readTimeout.toMillis.toInt)
-      .asBytes)
-  )
-
   private var _isOpen: Boolean = true
 
-  override def read(byteBuffer: ByteBuffer): Int =
-    responseBox match {
-      case Full(response) =>
-        val bytes = response.body
-
-        if (!response.isSuccess) {
-          val bodyString = new String(bytes, StandardCharsets.UTF_8)
-          throw new Exception(s"Https read failed for uri $uri: ${response.statusLine} – ${bodyString.take(1000)}")
-        }
-
-        val lengthToCopy = bytes.length - position
-        byteBuffer.put(bytes.drop(_position.toInt))
-        _position += lengthToCopy
-        lengthToCopy.toInt
-      case f: Failure =>
-        throw new Exception(s"Https read failed for uri $uri: $f")
-      case Empty =>
-        throw new Exception(s"Https read failed for uri $uri: Empty")
+  private lazy val backend = HttpClientSyncBackend(options = SttpBackendOptions.connectionTimeout(connectionTimeout))
+  private val rangeRequestsActive = openOptions.contains(ByteChannelOptions.RANGE)
+
+  private lazy val authenticatedRequest = path.getBasicAuthCredential.map { credential =>
+    basicRequest.auth.basic(credential.username, credential.password)
+  }.getOrElse(
+      basicRequest
+    )
+    .readTimeout(readTimeout)
+
+  private lazy val headRequest: Request[Either[String, String], Any] =
+    authenticatedRequest.head(Uri(uri))
+
+  private def getHeaderInformation = {
+    val response: Identity[Response[Either[String, String]]] = backend.send(headRequest)
```

Comment:
Do you know what’s the deal with the Either[String,String] here? Did you chose this or did it come from the library? What’s the semantics?

Reply:
It comes from the library, where the left side of the either is for error message and the right side for successful responses

Reply:
I see. Well, fair enough, thanks!